### PR TITLE
[WIP] Docstring review

### PR DESCRIPTION
**Pull request by BibMartin**
Starting to address #140 
Hunting `TODO` with `grep -r TODO`

### CODE REVIEW

**Comment by ocefpaf** [comment]
@andrewgiessel Can I recruit you to review these? (As the only native English speaker I guess you are the most qualified to review docstrings :stuck_out_tongue_closed_eyes:)

File: folium/element.py [hunk 1]
```diff
@@ -378,8 +380,22 @@ def add_subplot(self, x, y, n, margin=0.05):
...
-        """TODO : docstring here"""
+        """Create an HTML div object for embedding data.
+
+        Parameters
+        ----------
+        data : str
+            The HTML data to be embedded.
+        width : int or str, default '100%'
+            The width of the output div.
+            Ex : 120 , '120px', '80%'
```
Comment (ocefpaf) [inline folium/element.py]: We can drop the space between `Ex` and `:`

File: folium/element.py [hunk 2]
```diff
@@ -378,8 +380,22 @@ def add_subplot(self, x, y, n, margin=0.05):
...
+        height : int or str, default '100%'
+            The height of the output div.
```
Comment (ocefpaf) [inline folium/element.py]: `div element` (just a little bit more verbose)

File: folium/element.py [hunk 3]
```diff
@@ -107,7 +107,9 @@ def get_root(self):
             return self._parent.get_root()
 
     def render(self, **kwargs):
-        """TODO : docstring here."""
+        """Renders the Element into an HTML representation, using the `_template`
```
Comment (ocefpaf) [inline folium/element.py]: I don't think we need that comma.

File: folium/element.py [hunk 4]
```diff
@@ -325,7 +327,7 @@ def get_root(self):
         return self
 
     def render(self, **kwargs):
-        """TODO : docstring here."""
+        """Renders the Element into an HTML representation."""
```
Comment (ocefpaf) [inline folium/element.py]: How about `Renders the HTML representation of the element`?  (If you decide to change modify the docstring above too.)

File: folium/element.py [hunk 5]
```diff
@@ -378,8 +380,22 @@ def add_subplot(self, x, y, n, margin=0.05):
 
 
 class Html(Element):
+    """A basic Element for embedding HTML.
+    """
```
Comment (ocefpaf) [inline folium/element.py]: Make this a 1-line docstring.

File: folium/element.py [hunk 6]
```diff
@@ -378,8 +380,22 @@ def add_subplot(self, x, y, n, margin=0.05):
...
+        Parameters
+        ----------
+        data : str
+            The HTML data to be embedded.
+        width : int or str, default '100%'
+            The width of the output div.
+            Ex : 120 , '120px', '80%'
+        height : int or str, default '100%'
+            The height of the output div.
+            Ex : 120 , '120px', '80%'
```
Comment (ocefpaf) [inline folium/element.py]: Ditto

File: folium/element.py (hunk 4 above)
Comment (ocefpaf) [inline folium/element.py]: And the last as well.

File: folium/element.py (hunk 6 above)
Comment (BibMartin) [inline folium/element.py]: ???

Comment (ocefpaf) [inline folium/element.py]: I meant same as https://github.com/python-visualization/folium/pull/329/files#r49851692

`Ex :` &rarr; `Ex:`

**Comment by BibMartin** [comment]
```(suggestion + I agree) = correction ```

**Comment by BibMartin** [comment]
@ocefpaf 
At the cost of being here ; I would like to drop `features.ColorScale` object in favor of `colormap.ColorMap`, so that we don't have to handle future deprecation warnings.

What do you think ? Can I do it in this PR, or wait for it to be merged and do another one ?

**Comment by BibMartin** [comment]
Same question about fixing #247

**Comment by ocefpaf** [comment]
Do what is most convenient. I am OK with a mixed porpuse PR.

**Comment by BibMartin** [comment]
:white_check_mark: I think I'm done for this PR.
@ocefpaf Please tell me if you want I squash, if you're okay with all this mess.

**Comment by ocefpaf** [comment]
> :white_check_mark: I think I'm done for this PR.

Thanks!  I will take a second look and merge it.

> @ocefpaf Please tell me if you want I squash, if you're okay with all this mess.

Let's not squash this one in case we need to reference a certain commit later.

**Comment by ocefpaf** [comment]
A few extra minor comments and I am done.  (BTW.  I am OK with the docstings as is and I can merge without the suggestions if you do not want to change.)

**Comment by BibMartin** [comment]
Finally I removed the title to keep it simple.

If you're okay, it's ready to go.

**Comment by ocefpaf** [comment]
Awesome!

**Pull request merged**